tests/rbd/test_rbd_image_zero_size.py

```diff
@@ -0,0 +1,115 @@
+4. Use rbd du command to check that image size for both images
+5. Resize second image size to 0
+6. Remove that images
+
+"""
+
+from ceph.parallel import parallel
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
+    initial_rbd_config,
+    verify_image_feature_exist,
+    verify_image_size,
+)
+from ceph.rbd.workflows.rbd import resize_single_image_and_verify
+from utility.log import Log
+from utility.utils import run_fio
+from cli.rbd.rbd import Rbd
+import json
+
+log = Log(__name__)
+
+def resize_image_with_zero_size(rbd, rbd1, pool_type, **kw):
+    """
+        Test to verify RBD image with zero size.
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+    """
+
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        resize_to = kw["config"]["resize_to"]
+ 
+        if pool_type == "rep_pool_config": 
+            log.info(f"Resize Image {image} to zero \n\n")
+            resize_config = {"image-spec": f"{pool}/{image}", "size": resize_to, "allow-shrink": True}
+            out, err = rbd.resize(**resize_config)
+            if "100% complete" not in out + err:
+                log.error(
+                    f"Image {image} resize to 0 failed with error {out} {err}"
+                )
+                return 1
+
+        log.info(f"Verify Image {image} size")
+        info_spec = {"image-or-snap-spec": f"{pool}/{image}", "format": "json"}
+        out, err = rbd.info(**info_spec)
+        if err:
+            log.info(f"Error while fetching info for image {image}")
+            return 1
+        log.info(f"Image info: {out}")
+
+        out_json = json.loads(out)
+        if "0" not in str(out_json["size"]):
+            log.error(f"Image size Verification failed for {image}: Image size should be zero")
+            return 1
+
+        return 0
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        log.info("Cleanup")
+        if pool_type == "ec_pool_config":
+            rbd1.clean_up(pools=[kw["config"][pool_type]["pool"]])
+
+
+def run(**kw):
+    """Test to verify RBD image with zero size.
+    This test verifies images with zero size
+    Args:
+        kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    """
+    log.info("CEPH-83597243 -  Test to verify RBD image with zero size \n\n")
+    
+    rbd_obj = initial_rbd_config(**kw)
+    
+    if kw.get("client_node"):
+        client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+    else:
+        client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+    rbd = Rbd(client)
+
+    if rbd_obj:
+        log.info("Executing test on Replication pool \n\n")
+        if resize_image_with_zero_size(
```

Comment:
no need to pass generic `rbd` again here, can you please refer this test 
https://github.com/red-hat-storage/cephci/blob/master/tests/rbd/test_rbd_encryption.py#L16

Reply:
Done